src/anemoi/training/diagnostics/callbacks/plot.py

```diff
@@ -151,26 +154,33 @@ def _plot(
     ) -> None:
         """Plotting function to be implemented by subclasses."""
 
+    # Async function to run the plot function in the background thread
+    async def submit_plot(self, trainer, *args, **kwargs) -> None:
+        """Async function or coroutine to schedule the plot function."""
+        loop = asyncio.get_running_loop()
+        await loop.run_in_executor(
+            self._executor,
+            self._plot_with_error_catching,
+            trainer,
+            *args,
+            *kwargs.values(),
```

Comment:
This implies that kwargs will be provided in the correct order?
Why can't this be expanded normally?

Reply:
Or maybe something like this ?

```
def func(s, trainer, args, kwargs):
  return s._plot_with_error_catching(trainer, *args, **kwargs)
  
await loop.run_in_executor(
             self._executor,
             func,
             self,
             trainer,
             args,
             kwargs,
 ```
 
 I am not sure we need to pass self as an argument.

Reply:
So I had a better look into this. And what's happening is that the async function `run_in_executor` doesn't support keyword arguments, so using *kwargs.values() bypasses the need to pass keywords directly by transforming the keyword arguments into positional ones. In that way the values are then correctly used by `_plot_with_error_catching` as positional arguments, which matches the signature that accepts *args.
I have updated the code to reflect that! happy to discuss further!